vision_datasets/image_object_detection/detection_as_kvp_dataset.py

```diff
@@ -0,0 +1,113 @@
+
+        schema = self.construct_schema(self.class_names)
+        # Update dataset_info with schema
+        dataset_info = KeyValuePairDatasetInfo({**dataset_info_dict, "schema": schema})
+
+        # Construct KeyValuePairDatasetManifest
+        annotations = []
+        for id, img in enumerate(detection_dataset.dataset_manifest.images, 1):
+            bboxes = [box.label_data for box in img.labels]
+            # label_names = [self.class_id_to_names[box[0]] for box in bboxes]
+
+            kvp_label_data = self.construct_kvp_label_data(bboxes)
+            img_ids = [self.img_id_to_pos[img.id]]  # 0-based index
+            kvp_annotation = KeyValuePairLabelManifest(id, img_ids, label_data=kvp_label_data)
+
+            # KVPDatasetManifest expects img.labels to be empty. Labels are instead stored in KVP annotation
+            img.labels = []
+            annotations.append(kvp_annotation)
+
+        dataset_manifest = KeyValuePairDatasetManifest(detection_dataset.dataset_manifest.images, annotations, schema, additional_info=detection_dataset.dataset_manifest.additional_info)
+        super().__init__(dataset_info, dataset_manifest, dataset_resources=detection_dataset.dataset_resources)
+
+    def construct_schema(self, class_names: typing.List[str]) -> typing.Dict[str, typing.Any]:
+        schema: typing.Dict[str, typing.Any] = BASE_DETECTION_SCHEMA  # initialize with base schema
+        schema["fieldSchema"][f"{BBOXES_KEY}"]["items"]["classes"] = {c: {} for c in class_names}
+        return schema
+
+    def construct_kvp_label_data(self, bboxes: typing.List[typing.List[int]]):
+        """
+        Convert the detection dataset label_name to the desired format for KVP annnotation as defined by the BASE_DETECTION_SCHEMA.
+        E.g. {"fields": {"bboxes": {"value": [{"value": "class1", "groundings" : [[10,10,20,20]]},
+                                              {"value": "class2", "groundings" : [[0,0,20,20], [20,20,30,30]]}]
+                        "text": None}
+        """
+
+        label_wise_bboxes = self._sort_bboxes_label_wise(bboxes)
+
+        return {
+            f"{KeyValuePairLabelManifest.LABEL_KEY}": {
+                f"{BBOXES_KEY}": {
+                    f"{KeyValuePairLabelManifest.LABEL_VALUE_KEY}": [{f"{KeyValuePairLabelManifest.LABEL_VALUE_KEY}": key, f"{KeyValuePairLabelManifest.LABEL_GROUNDINGS_KEY}": value} for key, value in label_wise_bboxes.items()]
+                }
+            },
+            f"{KeyValuePairLabelManifest.TEXT_INPUT_KEY}": None
+        }
+
+    def _sort_bboxes_label_wise(self, bboxes: typing.List[typing.List[int]]):
+        """
+        Convert a list of bounding boxes to a dictionary with class name as key and list of bounding boxes as value.
+
+        E.g. [[1, 2, 3, 4, 5], [1, 2, 3, 4, 5], [2, 3, 4, 5, 6]] -> {"1": [[2, 3, 4, 5], [2, 3, 4, 5]], "2": [[3, 4, 5, 6]]}
+        """
+
+        label_wise_bboxes = {}
```

Comment:
I think this function should be assigning the names of the categories to the keys, not just the index. i.e., it should be using `self.class_id_to_names`. So for each entry in `label_wise_bboxes`, we would have the key be `self.class_id_to_names[box[0]]` be the key, not `str(box[0])`.

Reply:
Good catch! Fixed.